beacon_node/eth2-libp2p/src/behaviour/mod.rs

```diff
@@ -725,5 +892,6 @@ pub enum BehaviourEvent<TSpec: EthSpec> {
     /// Subscribed to peer for given topic
     PeerSubscribed(PeerId, TopicHash),
     /// Inform the network to send a Status to this peer.
+    /// TODO: why not just do it here?
```

Comment:
I was trying very hard to keep the behaviour and network separate from the consensus structures, like beacon chain. So the network could function freely, independent of whatever is going on at the application level. 
A Status request requires knowledge of our current state in the chain and fork etc. 
The original design was the `Network` crate had knowledge of this kind of information, whilst the libp2p network was entirely separate. 
We would need access to a `BeaconChain` to send a Status. Last I checked I think I was successful in keeping the beacon chain out of the behaviour, but I remember it kept trying to sneak back in. I think it's still not in at the moment